src/pages/index.js

```diff
@@ -62,36 +104,160 @@ function handleImageClick(cardData) {
 
 //edit profile
 function handleProfileSubmit(data) {
-  userInfo.setUserInfo({name: data.name, job: data.description});
-editPopup.close(); }
+  renderLoading(true, editPopup);
+  api.updateUserInfo({
+    name: data.name,
+    about: data.about
+  })
+    .then(updatedUserData => {
+      user.setUserInfo({
+        name: updatedUserData.name, 
+        about: updatedUserData.about || updatedUserData.dascription || updatedUserData.bio
+      });
+    })
+    .catch(err => console.error(`Error updating profile: ${err}`))
+    .finally(() =>{
+      renderLoading(false, editPopup);
+      editPopup.close();
+    }
+    )
+   }
 
-//add card
+// Add card
 function handleAddCardSubmit(data) {
-  const card = {
+  const cardData = {
     name: data.title,
     link: data.url
   };
-  const cardElement = createCard(card);
-  section.addItems(cardElement);
-  addcardPopup.close();
-  profileAddFormElement.reset();
-  addFormValidator.disableButton();
+  renderLoading(true, addCardPopup);
+  // Create the Card 
+  api.addNewCard(cardData)
+    .then((newCardData) => {
+      const cardElement = createCard(newCardData);
+      section.addItems(cardElement);  
+      addCardPopup.close();            
+      profileAddFormElement.reset();  
+      addFormValidator.disableButton();  
+    })
+    .catch(err => {
+      console.error(`Error adding card: ${err}`);
+    })
+    .finally(() => {
+      renderLoading(false, addCardPopup)
+    }
+    )
 }
 
-function openEditProfileModal() {
-  const currentUser = userInfo.getUserInfo();
+
+
+//avatar
+function handleAvatarSubmit(data){
+  renderLoading(true, avatarPopup);
+  api.updateUserAvatar({
+    avatar: data.url
+  })
+    .then(updateUserData => {
+      user.setUserInfo({
+        avatar: updateUserData.avatar
+      });
+      
+    })
+    .catch(err => console.error(`Error updating avatar: ${err}`))
+    .finally(() =>{
+      renderLoading(false, avatarPopup);
+      avatarPopup.close();
+      avatarFormElement.reset();
+    });
+}
+
+//delete
+function handleDeleteSubmit(card){
+  const confirmButton = document.querySelector('#delete-confirm-button');
+  deletePopup.open();
+  
+  confirmButton.addEventListener('click', function onConfirmClick() {
+    renderLoading(true, deletePopup);
+    card.removeCard();
```

Comment:
You need to **delete** cards only in a `then` block after a **successful** response from the server, so that if there is an error, the card should stay in the `DOM`.